src/Wrapper/PropertyAware.php

```diff
@@ -67,35 +82,28 @@ protected function getPropertyValue(PimProperty $property)
             case CastType::TEXT:
             case CastType::HTML:
                 /** @var PropertyValue|null $propertyValue */
-                $propertyValue = Translator::translate($property->getValues(), $this->config->getLanguage());
+                $propertyValue = Translator::translate(
+                    $property->getValues(),
+                    $this->config->getLanguage()
+                );
                 if ($propertyValue) {
                     return $propertyValue->getValue();
                 }
 
                 return null;
             case CastType::SELECTION:
-                if (!$property->getValues()) {
+                if (!$property->getValues() || !$this->registryService->getPropertySelections()) {
                     return null;
                 }
 
-                foreach ($property->getPropertyData()->getSelections() as $selection) {
-                    if ($property->getValues()[0]->getValue() != $selection->getId()) {
-                        continue;
-                    }
-
-                    $relation = $selection->getRelation();
-                    if (!$relation) {
-                        continue;
-                    }
+                $value = $this->registryService->getPropertySelections()->getPropertySelectionValues(
```

Comment:
We can be explicit about the naming, because `$value` isn't very descriptive. I need to check the method name in line 77 to know for what this value is used, but I would still have no idea where this is coming from, when I am checking this from line 106.

```suggestion
                $propertySelectionValues = $this->registryService->getPropertySelections()->getPropertySelectionValues(
```

This makes it also clearer, why we are using `reset($value)` at the end, as it contains an array of selection values.